base/missing.jl

```diff
@@ -114,6 +114,9 @@ for f in (:(ceil), :(floor), :(round), :(trunc))
         ($f)(::Type{T}, ::Missing) where {T} =
             throw(MissingException("cannot convert a missing value to type $T: use Union{$T, Missing} instead"))
         ($f)(::Type{T}, x::Any) where {T>:Missing} = $f(nonmissingtype(T), x)
+        # to fix ambiguities
+        ($f)(::Type{T}, x::Rational) where {T>:Missing} = $f(nonmissingtype(T), x)
+        ($f)(::Type{T}, x::Rational{Bool}) where {T>:Missing} = $f(nonmissingtype(T), x)
```

Comment:
Why are these definitions specific to `Rational`?   Which method definitions are conflicting here?

Reply:
cc @nalimilan … I don't really understand the `T:>Missing` here.  Isn't that just `T==Missing` or `T==Any`?  How is that different from `::Type`?

Reply:
[This is a existing piece of code.](https://github.com/JuliaLang/julia/blob/master/base/missing.jl#L118-L119) I added these [lines of code](https://github.com/JuliaLang/julia/blob/f51f7c4c43c41a845090966bfc06040055585498/base/rational.jl#L422-L427) to remove the ambiguity : 
```
  MethodError: round(::Type{Float64}, ::Rational{Bool}) is ambiguous. Candidates:
    round(::Type{T}, x::Rational) where T<:Real in Base at rational.jl:385
    round(::Type{T}, x::Rational{Bool}) where T<:Real in Base at rational.jl:416
    round(::Type{T}, x::Real) where T<:AbstractFloat in Base at float.jl:363
```